### PR TITLE
KubernetesConfiguration Fix SDK generation readmes to contain package names

### PR DESCRIPTION
**Pull request by jonathan-innis**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jonathan-innis** [comment]
Hi @ArcturusZhang @jsntcy @msyyc @qiaozha. We are trying to understand why the `.zip` file that is showing here 
image does not contain the latest version of the generated SDK from the CI pipeline. `.zip` file looks to have an older version of the python SDK

**Comment by jonathan-innis** [comment]
Also, I want to understand why the Java and Powershell SDK generation is failing

File: specification/kubernetesconfiguration/resource-manager/readme.md [hunk 1]
```diff
@@ -25,46 +25,59 @@ For other options on installation see [Installing AutoRest](https://aka.ms/autor
...
+
+These settings apply only when `--tag=package-extensions-preview-2020-07` is specified on the command line.
+
+```yaml $(tag) == 'package-extensions-preview-2020-07'
+input-file:
+  - Microsoft.KubernetesConfiguration/preview/2020-07-01-preview/extensions.json
+```
 
-### Tag: package-2019-11-01-preview
+### Tag: package-source-control-configurations-preview-2029-11
```
Comment (ArcturusZhang) [inline specification/kubernetesconfiguration/resource-manager/readme.md:76]:
```suggestion
### Tag: package-source-control-configurations-preview-2019-11
```

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
I notice that we are changing the `input-file` of the existing tags for some reason, could you please explain why?
Changing the `input-file` in a tag, especially removing files from a tag will cause massive breaking changes in downstream SDKs.

From POV of golang SDK, we require all the tags with go SDK generation could cover all of the functionalities in this RP unless there is an explicit exception.

**Comment by ArcturusZhang** [comment]
@ChenTanyi to take a look at the java SDK issues.

**Comment by jonathan-innis** [comment]
> I notice that we are changing the `input-file` of the existing tags for some reason, could you please explain why?
> Changing the `input-file` in a tag, especially removing files from a tag will cause massive breaking changes in downstream SDKs.
> 
> From POV of golang SDK, we require all the tags with go SDK generation could cover all of the functionalities in this RP unless there is an explicit exception.

We are trying to move from our old tagging pattern where we are specifying the api version for all resources released under that api version to having specific package tags for the combination of api version and ARM resource. The plan in the future is to release api versions that only apply to one resource and not the other. 

Also, we want to support multi-package support in SDKs like typescript where we want to use the latest version of api for source control configuration resource (2021-03-01) and the latest version for extension resource (2020-07-01-preview) together in the SDK package. 

@ArcturusZhang Do you have a recommendation on how to make these changes without introducing breaking changes into downstream sdks?

File: specification/kubernetesconfiguration/resource-manager/readme.md [hunk 2]
```diff
@@ -25,46 +25,59 @@ For other options on installation see [Installing AutoRest](https://aka.ms/autor
 These are the global settings for the KubernetesConfiguration.
 
 ``` yaml
-title: SourceControlConfigurationClient
+title: KubernetesConfigurationClient
 description: KubernetesConfiguration Client
 openapi-type: arm
-tag: package-2021-03
```
Comment (ChenTanyi) [inline specification/kubernetesconfiguration/resource-manager/readme.md:31]: Java Automation depends on default tag. Can any of the tag be the default tag?

**Pull request closed without merging**